#
# This provides padding for the krb5_keyblock struct so its size will be the
# same as it was in ON krb to avoid ABI issues that may be caused by the
# difference.  This difference was a result of modifying ON krb to use the
# Crypto Framework/libpkcs11 API as the crypto provider.  This patch is not
# intended for upstream contribution to MIT.
# Patch source: in-house
#

--- a/src/include/krb5/krb5.hin
+++ b/src/include/krb5/krb5.hin
@@ -362,6 +362,13 @@ typedef struct _krb5_keyblock {
     krb5_enctype enctype;
     unsigned int length;
     krb5_octet *contents;
+    /*
+     * Solaris Kerberos: for backwards ABI compatibility.  The following fields
+     * are not used by the MIT code:
+     */
+    int   *dk_list; /* krb5_dk_node */
+    unsigned long hKey; /* CK_OBJECT_HANDLE */
+    int pid;
 } krb5_keyblock;
 
 struct krb5_key_st;
